### PR TITLE
Fix for #498: handling "12pm-12am"

### PR DESCRIPTION
**Pull request by HubMiner**
- (---Done on top of "russian" change; tests pass; fixed one pre-existing bad UT.---)
- Rebased to master.

### CODE REVIEW

File: modules/oh_sanitizer/main.py [hunk 1]
```diff
@@ -461,6 +463,8 @@ def test_invalid_fields(self):
         self.assertEqual(sanitize_field("8h45 am - 11.45 a.m."), "08:45-11:45")
         self.assertEqual(sanitize_field("9h p.m. 6 - 10 pm 15"), "21:06-22:15")
         self.assertEqual(sanitize_field("9 am - 12+"), "09:00-12:00+")
+        self.assertEqual(sanitize_field("12:33 pm - 12:44 am"), "12:33-24:44")
```
Comment (frodrigo) [inline modules/oh_sanitizer/main.py]: `24:44` should not be `00:44` ?

File: modules/oh_sanitizer/main.py [hunk 2]
```diff
@@ -290,7 +290,9 @@ def hour_minutes(self, args):
         for arg in args:
             if arg.type == 'MINUTE':
                 m = int(arg)
-            elif arg.type == 'PM':
+            elif ( (arg.type == 'AM') and (h == 12) ):
```
Comment (frodrigo) [inline modules/oh_sanitizer/main.py]: Can you remove extra parenthesis, please?

**Comment by HubMiner** [comment]
* added code forcing 24+ value to (-24) + UTs

File: modules/oh_sanitizer/main.py [hunk 3]
```diff
@@ -290,8 +290,13 @@ def hour_minutes(self, args):
...
             if arg.type == 'MINUTE':
                 m = int(arg)
-            elif arg.type == 'PM':
-                h = h + 12
+            elif ( arg.type == 'AM' and h == 12 ):
+                h = 0
+            elif ( arg.type == 'PM' and h < 12 ):
+                h += 12
+        if (h >= 24):
+            h -= 24
```
Comment (frodrigo) [inline modules/oh_sanitizer/main.py]: Can you use a modulo, il all case: `h = h % 24`

File: modules/oh_sanitizer/main.py [hunk 4]
```diff
@@ -281,7 +281,7 @@ def hms(self, args):
         if len(args) > 1 and args[1].type == 'PM':
             h = h + 12
         return str(h).zfill(2) + ':' + str(m).zfill(2)
-
+
```
Comment (frodrigo) [inline modules/oh_sanitizer/main.py]: Sorry, but please revert this extra space.

**Comment by HubMiner** [comment]
Hi @frodrigo, please merge this.

**Comment by frodrigo** [comment]
Thank.

**Pull request merged**